src/main/java/io/vertx/core/tracing/TracingOptions.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2011-2017 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.tracing;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.spi.VertxTracerFactory;
+
+/**
+ * Vert.x tracing base configuration, this class can be extended by provider implementations to configure
+ * those specific implementations.
+ *
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+@DataObject(generateConverter = true, publicConverter = false)
+public class TracingOptions {
+
+  /**
+   * The default value of tracing enabled false
+   */
+  public static final boolean DEFAULT_TRACING_ENABLED = false;
+
+  private boolean enabled;
+  private JsonObject json; // Keep a copy of the original json, so we don't lose info when building options subclasses
+  private VertxTracerFactory factory;
+
+  /**
+   * Default constructor
+   */
+  public TracingOptions() {
+    enabled = DEFAULT_TRACING_ENABLED;
+  }
+
+  /**
+   * Copy constructor
+   *
+   * @param other The other {@link TracingOptions} to copy when creating this
+   */
+  public TracingOptions(TracingOptions other) {
+    enabled = other.isEnabled();
+    factory = other.factory;
+  }
+
+  /**
+   * Create an instance from a {@link JsonObject}
+   *
+   * @param json the JsonObject to create it from
+   */
+  public TracingOptions(JsonObject json) {
+    this();
+    TracingOptionsConverter.fromJson(json, this);
+    this.json = json.copy();
+  }
+
+  /**
+   * Will tracing be enabled on the Vert.x instance?
```

Comment:
define tracing and the impact. For example, does tracing imply header processing? If disabled will incoming headers propagated to outgoing requests? Some get confused by this.

Reply:
I think we can say that when tracing is enabled then header propagation would always happen and the reporting policy will depend on the implementation and its configuration, how does that sound ?